sound/soc/sof/intel/cnl.c

```diff
@@ -252,6 +252,19 @@ static bool cnl_compact_ipc_compress(struct snd_sof_ipc_msg *msg,
 int cnl_ipc4_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 {
 	struct sof_ipc4_msg *msg_data = msg->msg_data;
+	u32 hipcidr;
+	int ret;
+
+	ret = snd_sof_dsp_read_poll_timeout(sdev, HDA_DSP_BAR,
```

Comment:
@ujfalusi this makes me wonder if we have artificially doubled the IPC timeout for sending and IPC and getting a reply.
The reason I ask is lets say a previous IPC has not been acked by the FW and we're polling here waiting for it and the ack arrives just at the end of the polling time (which is the IPC timeout time), we'd go ahead and send the IPC and wait another ipc_timeout ms to get the reply, right? Should we roll up the send_msg into the original wait we have today instead?